docs/cpp/config-clang-mac.md

```diff
@@ -131,155 +150,189 @@ Next, we need to create a `tasks.json` file to tell VS Code how to build (compil
+The C/C++ extension attempts to populate `compilerPath` with the default compiler location based on what it finds on your system. The extension looks in several common compiler locations.
 
-    ![Breakpoint in main](images/cpp/breakpoint-in-main.png)
+The `compilerPath` search order is:
 
-    Then press `kb(workbench.action.debug.start)` to start execution from the current line in the standard library header. Execution will break on `cout`. If you like, you can press `kb(editor.debug.action.toggleBreakpoint)` again to toggle off the breakpoint.
+* Your PATH for the names of known compilers. The order the compilers appear in the list depend on your PATH.
+* Then hard-coded XCode paths are searched, such as `/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/`
 
-## Set a watch
+### Mac framework path
 
-Sometimes you might want to keep track of the value of a variable as your program executes. You can do this by setting a **watch** on the variable.
+On the C/C++ Configuration screen, scroll down and expand **Advanced Settings** and ensure that `Mac framework path` points to the system header files. For example: `/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks`
 
-1. Place the insertion point inside the loop. In the **Watch** window, click the plus sign and in the text box, type `word`, which is the name of the loop variable. Now view the Watch window as you step through the loop.
+## Reusing your C++ configuration
 
-   ![Watch window](images/cpp/watch-window.png)
+VS Code is now configured to use Clang on macOS. The configuration applies to the current workspace. To reuse the configuration, just copy the JSON files to a `.vscode` folder in a new project folder (workspace) and change the names of the source file(s) and executable as needed.
 
-1. Add another watch by adding this statement before the loop: `int i = 0;`. Then, inside the loop, add this statement: `++i;`. Now add a watch for `i` as you did in the previous step.
+## Troubleshooting
 
-1. To quickly view the value of any variable while execution is paused on a breakpoint, you can hover over it with the mouse pointer.
+### Compiler and linking errors
 
-   ![Mouse hover](images/cpp/mouse-hover.png)
+The most common cause of errors (such as undefined _main, attempting to link with file built for unknown-unsupported file format, and so on) occurs when `helloworld.cpp` is not the active file when you start a build or start debugging. This causes the compiler to try to compile something that isn't source code, like your launch.json, tasks.json, or c_cpp_properties.json file.
```

Comment:
Put _main in an inline code block `_main`.
Also launch.json, tasks.json, and c_cpp_properties.json should be wrapped in an inline code block. Example `launch.json`

Reply:
fixed